docs/concepts/true-hybrid-search.mdx

```diff
@@ -0,0 +1,266 @@
+  C --> E@{ shape: docs, label: "Top 100 Vector search results" }
+
+  D --> F{Merge and rerank top 200 results using RRF}
+  E --> F
+
+  F --> G@{ shape: stadium, label: "Final 100 ranked results" }
+```
+
+<br />
+
+This approach is fundamentally **probabilistic** - the final "top 100" results are not guaranteed to be the actual best matches because some potential matches might be missed if they don't appear in either index's top results.
+
+**TopK is different.** We scan both vector and keyword indexes simultaneously within a single query, ensuring that our "top 100" results are the **actual** top 100 - not just a probabilistic approximation:
+
+```mermaid
+flowchart TD
+    A[User Query]
+    A --> B[(Keyword Index)]
+    A --> C[(Vector Index)]
+
+    B --> D{Simultaneously scan both indexes and retrieve top 100 results}
+    C --> D
+    D --> E@{ shape: docs, label: "Combined top 100 Keyword and Vector search results" }
+    E --> F@{ shape: stadium, label: "Final 100 ranked results" }
+```
+
+
+With TopK, you can:
+
+- Retrieve documents based on multiple embeddings -- **multi-vector retrieval**
+- Combine semantic similarity (or vector distance) with keyword search -- **true hybrid retrieval**
+- Filter documents to enforce hard constraints -- **metadata filtering**
+- Apply custom scoring functions blending multiple ranking factors -- **custom scoring**
+
+
+
+## Implementing Hybrid Search(Vector + Keyword)
+
+Hybrid retrieval combines **semantic similarity (vector-based search)** with **exact keyword matching**. This ensures that documents containing **explicit matches** to the query keywords are considered alongside **semantic similarity**.
+
+Let's define a collection with one [`keyword_index()`](/collections/create#keyword-index) and one [`semantic_index()`](/collections/create#semantic-index):
+
+<CodeGroup>
+
+```python Python
+from topk_sdk.schema import text, semantic_index, keyword_index
+
+client.collections().create(
+    "articles",
+    schema={
+        "title": text().required().index(keyword_index()),  # Keyword-based retrieval
+        "content": text().index(semantic_index()),  # Semantic search
+    },
+)
+```
+
+
+```typescript Javascript
+import { text, semanticIndex, keywordIndex } from "topk-js/schema";
+
+await client.collections().create("articles", {
+  title: text().required().index(keywordIndex()), // Keyword-based retrieval
+  content: text().index(semanticIndex()), // Semantic search
+});
+```
+
+</CodeGroup>
+
+In the following example we'll perform a hybrid search that scans both keyword and vector(semantic) index:
+
+<CodeGroup>
+
+```python Python
+from topk_sdk.query import select, fn, match
+
+docs = client.collection("articles").query(
+    select(
+        "title",
+        content_similarity=fn.semantic_similarity("content", "climate change policies"),
+        text_score=fn.bm25_score()
+    )
+    .filter(match("carbon tax") | match("renewable energy"))  # Ensure keyword relevance
```

Comment:
btw: matching on "carbon tax" will match on "carbon" and then "tax". matching on "tax" is somewhat more generic. I think what you intuitively wanted to do here is "phrase search" -- which we don't support just yet (or ever?)

Reply:
changed it to just `"carbon"` for now